nodes/rotation/direction_to_rotation.py

```diff
@@ -28,36 +31,18 @@ def draw(self, layout):
             layout.label("Must be different", icon = "ERROR")
 
     def getExecutionCode(self):
-        if self.trackAxis[-1:] == self.guideAxis[-1:]:
-            yield "eulerRotation = mathutils.Euler((0, 0, 0), 'XYZ')"
-            return
-
-        yield "zero = mathutils.Vector((0, 0, 0))"
-
-        yield "if direction == zero: eulerRotation = mathutils.Euler((0, 0, 0), 'XYZ')"
-        yield "else:"
-        yield "    z = direction.normalized()"
-        yield "    if guide != zero and z.cross(guide) != zero: y = z.cross(guide.normalized())"
-        if "X" == self.guideAxis: yield "    else: y = z.cross(mathutils.Vector((1, 0, 0))) if z.cross(mathutils.Vector((1, 0, 0))) != zero else mathutils.Vector((0, 0, 1))"
-        if "Y" == self.guideAxis: yield "    else: y = z.cross(mathutils.Vector((0, 1, 0))) if z.cross(mathutils.Vector((0, 1, 0))) != zero else mathutils.Vector((0, 0, 1))"
-        if "Z" == self.guideAxis: yield "    else: y = z.cross(mathutils.Vector((0, 0, 1))) if z.cross(mathutils.Vector((0, 0, 1))) != zero else mathutils.Vector((0, 1, 0))"
-
-        yield "    x = y.cross(z)"
-
-        yield "    mx, my, mz = " + getAxesChange(self.trackAxis, self.guideAxis)
-
-        yield "    mat3x3 = mathutils.Matrix().to_3x3()"
-        yield "    mat3x3.col[0], mat3x3.col[1], mat3x3.col[2] = mx, my, mz"
-        yield "    eulerRotation = mat3x3.to_euler()"
+        isLinked = self.getLinkedOutputsDict()
+        if not any(isLinked.values()): return ""
+        
+        rot = "eulerRotation" if isLinked["eulerRotation"] else ""
+        quat = "quaternionRotation" if isLinked["quaternionRotation"] else ""
+        mat = "matrixRotation" if isLinked["matrixRotation"] else ""
+
+        return generateDirectionToRotationCode("direction", "guide", 
+                                                self.trackAxis, self.guideAxis, 
+                                                matrixOutputName = "{}".format(mat), 
```

Comment:
cause I need to normalize it for matrix, output, else the scale is affected too
only for that